include/io.tcc

```diff
@@ -34,7 +34,7 @@ void IO<Tdim>::write_material_points(
 //! /tparam Tdim dimension
 template <unsigned Tdim>
 void IO<Tdim>::write_stresses(
-    const std::vector<std::array<double, Tdim * 2>>& stresses) {
+    const std::vector<std::shared_ptr<Point<Tdim>>>& materialpoints) {
```

Comment:
Iterate over `materialpoints` create a vector of stresses and pass it to this function.